devicetypes/smartthings/zwave-lock-without-codes.src/zwave-lock-without-codes.groovy

```diff
@@ -405,6 +408,11 @@ private def handleBatteryAlarmReport(cmd) {
 	def deviceName = device.displayName
 	def map = null
 	switch (cmd.zwaveAlarmEvent) {
+		case 0x01: //power has been applied, check if the battery level updated
+			state.queryBattery = true
+			result << "delay 1200"
```

Comment:
What is significance of 1200 milliseconds?

Also, `handleBatteryAlarmReport` is handled where the message is received, here we are setting `queryBattery` to true and then we are generating another battery get because the flag is set in line 213. So essentially sending two battery gets.

Reply:
Yale told us that the device just doesn't respond after batteries are changed. The ideal solution is that if we just wait a second to query the battery we won't have to do it again later.

I see your point about the flag. Any suggestion?